### PR TITLE
GH-951 Convert Sample Site Pages to Data Factory Test Data

### PR DESCRIPTION
**Pull request by dtsnell4**
Removes all references to old testData() method and instantiates all widgets using the new Factory data structure.

Fixes GH-951

Signed-off-by: Dan Snell <Dan.Snell@lexisnexis.com>

### CODE REVIEW

**Comment by dtsnell4** [comment]
@jbrundage @mlzummo Please review.  There is obviously a lot of code here, but please pay special attention to the demos themselves, especially ones you developed, and make sure they are functioning as expected.  http://rawgit.com/dtsnell4/Visualization/GH-951_SampleSite_Demos/apps/SampleSite/playground.html

**Comment by dtsnell4** [comment]
When we are good with it I will invite Gordon to review and pull.

**Comment by mlzummo** [comment]
amchart column = bar

File: apps/SampleSite/js/themeeditor.js [hunk 1]
```diff
@@ -564,4 +592,47 @@ function g_defaultSerials(idx,doReset) {
     } else {
         localStorage.themeEditorSerials = JSON.stringify(defaultSerials);
     }
+}
+function getProperName(name) {
```
Comment (mlzummo) [inline apps/SampleSite/js/themeeditor.js]: this function is a duplicate of another one used above ... can consolidate ....

File: apps/SampleSite/js/widget_index.js [hunk 2]
```diff
@@ -281,3 +159,64 @@ function afterGridLoads() {
         },1000);
     }
 }
+
+function getProperName(name) {
```
Comment (mlzummo) [inline apps/SampleSite/js/widget_index.js]: this function is a duplicate of 2 others above ... can consolidate

File: test/otherFactory.js [hunk 3]
```diff
@@ -17,19 +17,19 @@
...
-                        return [d, 10 + Math.random() * 14];
-                    });
-                    callback(new WordCloud()
-                        .columns(DataFactory.WordCloud.simple.columns)
-                        .data(words)
-                    );
-                });
-            }
-        },
+        // WordCloud: {
```
Comment (mlzummo) [inline test/otherFactory.js]: how come word cloud is removed .. if its because its not working ... i have that fixed in several PRs that are now passing travis ...

File: apps/SampleSite/PopupSamples/modal.html [hunk 4]
```diff
@@ -5,12 +5,32 @@
...
                 var modal = new Popup()
                     .size({width:500,height:500})
                     .target("main")
                     .position("fixed")
-                    .widget(new Surface().title("Modal Dialog Example").widget(new Polar().testData()).surfaceBackgroundColor("rgb(234, 249, 255)"))
+                    .widget(new Surface()
+                        .title("Modal Dialog Example")
+                        .surfaceBackgroundColor("rgb(234, 249, 255)")
+                        .widget(new Polar()
+                            .columns(["Subject", "Year 1", "Year 2", "Year 3", "Year 4"])
```
Comment (mlzummo) [inline apps/SampleSite/PopupSamples/modal.html]: this should be read from the dataFactory?

File: apps/SampleSite/PopupSamples/modalx.html [hunk 5]
```diff
@@ -5,14 +5,23 @@
...
                     .size({width:600,height:500})
                     .target("main")
                     .position("fixed")
                     .widget(new Surface()
                         .title("Modal Dialog With Close Button")
-                        .widget(new Gauge().testData())
+                        .surfaceBackgroundColor("rgb(234, 249, 255)")
+                        .widget(new Line()
+                            .columns(["Subject", "Year 1", "Year 2", "Year 3"])
+                            .data( [
```
Comment (mlzummo) [inline apps/SampleSite/PopupSamples/modalx.html]: this should be read from the dataFactory?

File: apps/SampleSite/PopupSamples/modeless.html [hunk 6]
```diff
@@ -24,6 +23,16 @@
                                 font: "FontAwesome"
                             }
                         ])
+                        .widget(new Column()
+                            .columns(["Subject", "Year 1", "Year 2", "Year 3"])
+                            .data( [
+                                ["Geography", 75, 68, 65],
```
Comment (mlzummo) [inline apps/SampleSite/PopupSamples/modeless.html]: this should be read from the dataFactory?

File: apps/SampleSite/dynamic_grid.html [hunk 7]
```diff
@@ -35,14 +35,24 @@
...
+                "title":"HPCC Column",
                 "src":[
-                    "src/amchart/Bar"
+                    "src/chart/Column"
                 ],
                 "buttons":[
                     {"type":"button", "name":"button1", "label":"button1", "value": "Button 1", "class":"standard_button", "title": "This is a standard button element"},
                     {"type":"button", "name":"button2", "label":"button2", "value": "Button 2", "class":"standard_button", "title": "This is also a standard button element"}
+                ],
+                "data":[[
```
Comment (mlzummo) [inline apps/SampleSite/dynamic_grid.html]: this should be read from the dataFactory?

**Comment by mlzummo** [comment]
popup page exmaples dont work

File: test/otherFactory.js (hunk 3 above)
Comment (dtsnell4) [inline test/otherFactory.js]: No, I commented it out to test something and forgot to uncomment, thanks

File: apps/SampleSite/js/themeeditor.js (hunk 1 above)
Comment (dtsnell4) [inline apps/SampleSite/js/themeeditor.js]: I agree but they have no common js files...  Should I make a new js file to include with just this?  Didn't seem like enough code, although moving forward it will be useful for new pages that have menus of all widgets...  Perhaps even creating a single menu-building script that can just be included in any new page

File: apps/SampleSite/PopupSamples/modal.html (hunk 4 above)
Comment (dtsnell4) [inline apps/SampleSite/PopupSamples/modal.html]: Per our conversation yesterday, its better for the users if they can see the data/column functions and be able to play with the data.

**Comment by dtsnell4** [comment]
@mlzummo Can you elaborate on the popup examples not working, they seem to be working for me...

**Comment by mlzummo** [comment]
@dtsnell4 its working now ... prolly rawgit didnt sync completely when i tried

File: demos/dermatology.html [hunk 8]
```diff
@@ -375,6 +375,7 @@
                 ;
                 if (first) {
                     first = false;
+
```
Comment (GordonSmith) [inline demos/dermatology.html]: Remove this change.

File: src/layout/Popup.js [hunk 9]
```diff
@@ -50,11 +50,7 @@
         HTMLWidget.prototype.update.apply(this, arguments);
         var context = this;        
 
-        if (this.popupState()) {
-            element.style("display", "block");
-        } else {
-            element.style("display", "none");
-        }
+        this.visible(this.popupState());
```
Comment (GordonSmith) [inline src/layout/Popup.js]: This change should be in a different PR.

Comment (dtsnell4) [inline src/layout/Popup.js]: @GordonSmith Yes, I was going to bring this up to you once the whole thing was ready for your review, I'm still working on things from J and Matts comments.  Should I make an issue for the Popup widget change for the PR to reference?

Comment (GordonSmith) [inline src/layout/Popup.js]: Please do yes.

**Comment by GordonSmith** [comment]
@dtsnell4 Needs rebase - @jbrundage @mlzummo Please review

**Comment by dtsnell4** [comment]
This is not ready I lost code in the rebase

**Comment by dtsnell4** [comment]
@mlzummo @jbrundage This is ready for review.  Please pay special attention to the pages you developed that they are working as expected.  

http://rawgit.com/dtsnell4/Visualization/GH-951_SampleSite_Demos/apps/SampleSite/playground.html

cc: @GordonSmith There should be no changes to the src folder here, but I did do some re-arranging of some test Factory data.  In the initial implementation there were some inconsistencies between different factories, particularly when a widget created another widget internally (for example Surface or Grid widgets).  In some cases they were created in the charts Factory file, in others the dataFactory.js.  I didn't realize it at the time I was creating them.  I updated them all to be created in their respective Factory file.  If that should be a separate PR let me know.

**Comment by jbrundage** [comment]
It looks good to me.

Note: There are quite a few CSS issues and issues... and issues with example code popups, but those are likely pre-existing and can be handled as separate items.

File: demos/dermatology.html [hunk 10]
```diff
@@ -65,6 +65,25 @@
...
+
+
+
+
+
+
+
+
+    <script src="../apps/SampleSite/js/example_code.js"></script>
```
Comment (GordonSmith) [inline demos/dermatology.html]: Remove bad formatting

File: hello.html [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+                    .size({width:400,height:400})
+                    .target("widget-wrapper")
+                    .data({ vertices: vertices, edges: edges })
+                    .layout("Hierarchy")
+                    .render()
+                ;
+            });
+        </script>
+        <div id="widget-wrapper"></div>
+</html>
```
Comment (GordonSmith) [inline hello.html]: Is this file meant to be included (its in the root folder)?

Comment (dtsnell4) [inline hello.html]: Definitely not, thank you.

**Comment by dtsnell4** [comment]
@GordonSmith updated based on comments

File: demos/dermatology.html [hunk 12]
```diff
@@ -65,6 +65,8 @@
     </style>
     <script src="http://rawgit.com/jrburke/requirejs/2.1.17/require.js"></script>
     <script src="../src/config.js"></script>
+    <script src="../apps/SampleSite/js/example_code.js"></script>
+
```
Comment (GordonSmith) [inline demos/dermatology.html]: Is this needed?

File: demos/dermatology.html [hunk 13]
```diff
@@ -191,6 +193,7 @@
                 thisWidget = "";
                 
             categoryOptions.each(function (d, idx) {
+                var cat = d;
```
Comment (GordonSmith) [inline demos/dermatology.html]: Is this needed?

File: demos/dermatology.html [hunk 14]
```diff
@@ -380,6 +383,7 @@
                 if (first) {
                     first = false;
                     testWidget(widgetTest ? testFactory.widgets[widgetPath][widgetTest].factory : d3.map(testFactory.widgets[widgetPath]).values()[0].factory, params);
+
```
Comment (GordonSmith) [inline demos/dermatology.html]: I don't think dermatology should change as part of this PR?

File: test/otherFactory.js [hunk 15]
```diff
@@ -17,19 +17,19 @@
...
+        //             var words = DataFactory.WordCloud.simple.words.map(function (d) {
+        //                 return [d, 10 + Math.random() * 14];
+        //             });
+        //             callback(new WordCloud()
+        //                 .columns(DataFactory.WordCloud.simple.columns)
+        //                 .data(words)
+        //             );
+        //         });
+        //     }
+        // },
```
Comment (GordonSmith) [inline test/otherFactory.js]: Remove / uncomment...

File: demos/dermatology.html (hunk 14 above)
Comment (dtsnell4) [inline demos/dermatology.html]: Correct, this was test code I missed when removing, thank you.

**Comment by dtsnell4** [comment]
@GordonSmith updated

File: test/mapFactory.js [hunk 16]
```diff
@@ -10,7 +10,7 @@
         ChoroplethCounties: {
             simple: function (callback) {
                 require(["test/DataFactory", "src/map/ChoroplethCounties"], function (DataFactory, ChoroplethCounties) {
-                    var rawData = DataFactory.Counties.simple.rawData;
+                    var rawData = (DataFactory.Counties.simple.rawData);
                     var countyData = rawData.map(function (item) {
```
Comment (GordonSmith) [inline test/mapFactory.js]: remove parens

File: test/mapFactory.js [hunk 17]
```diff
@@ -29,7 +29,7 @@
                         var item = countries.countryNames[key];
                         nameCodeMap[item.name] = key;
                     }
-                    var rawData = DataFactory.Countries.simple.rawData;
+                    var rawData = (DataFactory.Countries.simple.rawData);
```
Comment (GordonSmith) [inline test/mapFactory.js]: remove parens

File: demos/dermatology.html [hunk 18]
```diff
@@ -65,6 +65,7 @@
     </style>
     <script src="http://rawgit.com/jrburke/requirejs/2.1.17/require.js"></script>
     <script src="../src/config.js"></script>
+
```
Comment (GordonSmith) [inline demos/dermatology.html]: Remove whitespace edit

**Comment by dtsnell4** [comment]
@GordonSmith done
